src/scitiff/io.py

```diff
@@ -3,78 +3,115 @@
     else:
-        _metadata = _extract_metadata_from_datagroup(dg)
+        raise NotImplementedError(
+            "Extracting metadata from DataGroup to SCITIFF is not yet implemented."
+        )
 
-    return {'scitiffmeta': _metadata}
+    return SciTiffMetadataContainer(scitiffmeta=SciTiffMetadata(image=_metadata))
 
 
-def _export_data_array(da: sc.DataArray, file_path: str | pathlib.Path) -> None:
-    metadata = _extract_metadata_from_dataarray(da)
-    # Make sure the data is consistent with the metadata
+def _validate_dimensions(da: sc.DataArray) -> None:
+    if illegal_dim := [
+        dim for dim in da.dims if dim not in SCITIFF_IMAGE_STACK_DIMENSIONS
+    ]:
+        raise ValueError(
+            f"DataArray has unexpected dimensions: {','.join(illegal_dim)}. "
+            f"Allowed dimensions are: {SCITIFF_IMAGE_STACK_DIMENSIONS} "
+            "Use `scipp.DataArray.rename_dims` to rename the dimensions."
+        )
+
+
+def to_scitiff_image(da: sc.DataArray) -> sc.DataArray:
+    _validate_dimensions(da)
+    default_sizes = {"x": 1, "y": 1, "z": 1, "t": 1, "c": 1}
+    final_sizes = {**default_sizes, **da.sizes}
+    # Order of the dimensions is according to the HyperStacks tiff format.
+    order = SCITIFF_IMAGE_STACK_DIMENSIONS  # ("c", "t", "z", "y", "x")
+    final_sizes = {key: final_sizes[key] for key in order if key in final_sizes}
+    dims = tuple(final_sizes.keys())
+    shape = tuple(final_sizes.values())
+    sizes: dict[str, int] = dict(zip(dims, shape, strict=True))
     # It is because ``z`` dimension and ``c`` dimension are often not present
     # but it is require by the HyperStacks and scitiffmeta schema.
     # Also, HyperStacks require specific order of dimensions.
-    dims = metadata['data']['dims']
-    shape = metadata['data']['shape']
-    sizes = dict(zip(dims, shape, strict=True))
-    final_image = sc.broadcast(da, sizes=sizes)
+    return sc.broadcast(da, sizes=sizes)
+
+
+def _export_data_array(da: sc.DataArray, file_path: str | pathlib.Path) -> None:
+    final_image = to_scitiff_image(da)
+    metadata = extract_metadata(final_image)
     tf.imwrite(
         file_path,
         final_image.values,
         imagej=True,
-        metadata=json.dumps(metadata),
+        metadata={
+            key: json.dumps(value)
+            for key, value in metadata.model_dump(mode="json").items()
+            # Tiff metadata will automatically be translated to json-like-string
+            # so we need to convert the metadata to string in advance
+            # to catch the error early and make sure it can be loaded back.
+        },
         dtype=str(final_image.dtype),
     )
 
 
 def _export_data_group(dg: sc.DataGroup, file_path: str | pathlib.Path) -> None:
-    raise NotImplementedError('Exporting DataGroup to SCITIFF is not yet implemented.')
+    raise NotImplementedError("Exporting DataGroup to SCITIFF is not yet implemented.")
 
 
 def export_scitiff(
```

Comment:
For symmetry with `load_scitiff`, this should be called `save_scitiff`. (And the same for related functions.)